### PR TITLE
fix: remote builder: ssh-ng capability + builder user

### PR DESCRIPTION
**Pull request by DavHau**

### CODE REVIEW

File: nixos/roles/nix-remote-builder.nix [hunk 1]
```diff
@@ -27,10 +27,13 @@ in
     ];
 
     # Give restricted SSH access to the build scheduler
-    users.extraUsers.root.openssh.authorizedKeys.keys = map
+    users.extraUsers.builder.openssh.authorizedKeys.keys = map
```
Comment (zimbatm) [inline nixos/roles/nix-remote-builder.nix:30]:
```suggestion
    users.extraUsers.nix-remote-builder.openssh.authorizedKeys.keys = map
```
What do you think to remove ambiguity?

File: nixos/roles/nix-remote-builder.nix [hunk 2]
```diff
@@ -27,10 +27,13 @@ in
...
       (key:
-        ''command="nix-store --serve --write",no-agent-forwarding,no-port-forwarding,no-pty,no-user-rc,no-X11-forwarding ${key}''
+        ''command="nix-daemon --stdio",no-agent-forwarding,no-port-forwarding,no-pty,no-user-rc,no-X11-forwarding ${key}''
       )
       cfg.schedulerPublicKeys;
+    users.users.builder.isNormalUser = true;
+    users.users.builder.group = "nogroup";
+    nix.settings.trusted-users = ["builder"];
```
Comment (zimbatm) [inline nixos/roles/nix-remote-builder.nix:37]:
```suggestion
    users.users.nix-remote-builder.isNormalUser = true;
    users.users.nix-remote-builder.group = "nogroup";
    nix.settings.trusted-users = ["nix-remote-builder"];
```

**Review by zimbatm — COMMENTED** [review]
sounds good otherwise

**Comment by DavHau** [comment]
Still working on resolving GC issues. Will open another PR with fixes once we figured out the problem.

**Comment by DavHau** [comment]
bors merge

**Pull request merged**